crates/sdk/src/network/prover.rs

```diff
@@ -0,0 +1,345 @@
+                as u32;
+        let proof_results_path = env::var("PROOF_RESULTS_PATH").unwrap_or(".".to_string());
+        let asset_url = env::var("ASSET_URL").unwrap_or("http://152.32.186.45:20001".to_string());
+
+        let private_input = stdin.buffer.clone();
+        let mut pri_buf = Vec::new();
+        bincode::serialize_into(&mut pri_buf, &private_input)
+            .expect("private_input serialization failed");
+        let mut receipts = Vec::new();
+        let proofs = stdin.proofs.clone();
+        // todo: adapt to proof network after its updating
+        for proof in proofs {
+            let mut receipt = Vec::new();
+            bincode::serialize_into(&mut receipt, &proof)
+                .expect("private_input serialization failed");
+            receipts.push(receipt);
+        }
+
+        let prover_input = ProverInput {
+            elf: elf.to_vec(),
+            shard_size,
+            // execute_only,
+            composite_proof,
+            proof_results_path,
+            private_inputstream: pri_buf,
+            receipts,
+            asset_url,
+            ..Default::default()
+        };
+
+        log::info!("calling request_proof.");
+        let proof_id = self.request_proof(&prover_input).await?;
+
+        log::info!("calling wait_proof, proof_id={proof_id}");
+        let (proof, public_values) = self.wait_proof(&proof_id, timeout, &prover_input).await?;
+        Ok(ZKMProofWithPublicValues {
+            proof,
+            public_values,
+            stdin,
+            zkm_version: ZKM_CIRCUIT_VERSION.to_string(),
+        })
+    }
+}
+
+#[async_trait]
+impl Prover<DefaultProverComponents> for NetworkProver {
+    fn id(&self) -> ProverType {
+        ProverType::Network
+    }
+
+    fn zkm_prover(&self) -> &ZKMProver<DefaultProverComponents> {
+        self.local_prover.zkm_prover()
+    }
+
+    fn setup(&self, elf: &[u8]) -> (ZKMProvingKey, ZKMVerifyingKey) {
+        self.local_prover.setup(elf)
+    }
+
+    fn prove<'a>(
+        &'a self,
+        pk: &ZKMProvingKey,
+        stdin: ZKMStdin,
+        _opts: ProofOpts,
+        _context: ZKMContext<'a>,
+        _kind: ZKMProofKind,
+    ) -> Result<ZKMProofWithPublicValues> {
+        block_on(self.prove(&pk.elf, stdin, None))
+    }
+}
+
+async fn get_cert_and_identity(
+    ca_cert_path: &str,
+    cert_path: &str,
+    key_path: &str,
+) -> anyhow::Result<(Option<Certificate>, Option<Identity>)> {
+    let ca_cert_path = Path::new(ca_cert_path);
+    let cert_path = Path::new(cert_path);
+    let key_path = Path::new(key_path);
+    if !ca_cert_path.is_file() || !cert_path.is_file() || !key_path.is_file() {
+        bail!("both ca_cert_path, cert_path and key_path should be valid file")
+    }
+    let mut ca: Option<Certificate> = None;
+    let mut identity: Option<Identity> = None;
+    if ca_cert_path.is_file() {
+        let ca_cert = tokio::fs::read(ca_cert_path)
```

Comment:
we don't need to use async io here.  the entire network_prover::new can be sync.